src/intervals/rounding.jl

```diff
@@ -81,81 +85,80 @@ end
-    split_interval_string(Float64, string(x))
+function convert(::Type{Interval{Float64}}, x::Float64)
+    #isinf(x) && return Interval(x)
+    convert(Interval{Float64}, rationalize(x))
 end
 
-function make_interval(::Type{Float64}, x::Integer)
-    a = setprecision(53) do
-        make_interval(BigFloat, x)
-    end
-    float(a)
+convert(::Type{Interval{Float64}}, x::Rational) = @thin_round(float(typeof(x)), Float64(x))
+# round using the correct floating-point type
+
+function convert(::Type{Interval{Float64}}, x::Integer)
+    Interval( Float64(x, RoundDown), Float64(x, RoundUp) )
 end
 
-make_interval(::Type{Float64}, x::BigFloat) = @thin_round(Float64, convert(Float64, x))
 
-function make_interval(::Type{Float64}, x::Interval)
-    # a = make_interval(Float64, x.lo)
-    # b = make_interval(Float64, x.hi)
 
-    Interval( Float64(x.lo, RoundDown), Float64(x.hi, RoundUp) )
+convert(::Type{Interval{Float64}}, x::BigFloat) = @thin_round(BigFloat, Float64(x))
+# NB: Must use rounding of BigFloat, not of Float64, when converting BigFloats
 
+function convert(::Type{Interval{Float64}}, x::Interval)
+    Interval( Float64(x.lo, RoundDown), Float64(x.hi, RoundUp) )
 end
 
 
-# make_interval for Rational intervals
-function make_interval(::Type{Rational{Int}}, x::Irrational)
-    a = float(make_interval(BigFloat, x))
-    make_interval(Rational{Int}, a)
+# For complex:
+convert{T}(::Type{Interval{T}}, x::Complex{Bool}) = (x == im) ?
+    one(T)*im : throw(ArgumentError("Complex{Bool} not equal to im"))
+
+
+
+function convert(::Type{Interval{Rational{Int}}}, x::Irrational)
+    a = float(convert(Interval{BigFloat}, x))
+    convert(Interval{Rational{Int}}, a)
 end
-function make_interval(::Type{Rational{BigInt}}, x::Irrational)
-    a = make_interval(BigFloat, x)
-    make_interval(Rational{BigInt}, a)
+
+function convert(::Type{Interval{Rational{BigInt}}}, x::Irrational)
+    a = convert(Interval{BigFloat}, x)
+    convert(Interval{Rational{BigInt}}, a)
 end
-make_interval{T<:Integer, S<:Integer}(::Type{Rational{T}}, x::S) =
+
+convert{T<:Integer, S<:Integer}(::Type{Interval{Rational{T}}}, x::S) =
     Interval(x*one(Rational{T}))
-make_interval{T<:Integer, S<:Integer}(::Type{Rational{T}}, x::Rational{S}) =
+
+convert{T<:Integer, S<:Integer}(::Type{Interval{Rational{T}}}, x::Rational{S}) =
     Interval(x*one(Rational{T}))
-make_interval{T<:Integer, S<:Float64}(::Type{Rational{T}}, x::S) =
+
+convert{T<:Integer, S<:Float64}(::Type{Interval{Rational{T}}}, x::S) =
     Interval(rationalize(T, x))
-make_interval{T<:Integer, S<:BigFloat}(::Type{Rational{T}}, x::S) =
+
+convert{T<:Integer, S<:BigFloat}(::Type{Interval{Rational{T}}}, x::S) =
     Interval(rationalize(T, x))
-function make_interval{T<:Integer}(::Type{Rational{T}}, x::Interval)
-    a = make_interval(Rational{T}, x.lo)
-    b = make_interval(Rational{T}, x.hi)
-    Interval(a.lo, b.hi)
-end
 
 
 doc"""`transform` transforms a string by applying the function `f` and type
```

Comment:
Probably a better place to have `make_interval` and `transform` is in `src/macros`, since they are linked to `@interval` and its relatives.